org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/testexecution/TestExecutionCallTree.xtend

```diff
@@ -40,13 +42,33 @@ class TestExecutionCallTree {
 
 	def String getNodeJson(TestExecutionKey executionKey) {
 		val test = executionKey.testNode
-		val node = test.typedMapGetArray("children")?.findNode(executionKey.callTreeId)
+		val node = test.typedMapGetArray(childrenKey)?.findNode(executionKey.callTreeId)
 		if (node !== null) {
 			return node.writeToJsonHidingChildren
 		} else {
 			throw new IllegalArgumentException('''TestExecutionKey = '«executionKey»' cannot be found in call tree.''')
 		}
 	}
+	
+	def Iterable<TestExecutionKey> getChildKeys(TestExecutionKey key) {
```

Comment:
we should avoid 'child' in this context, since it  is a whole tree with children, grandchildren ...

Reply:
as discussed, I kept the name of both methods, and instead refactored the "inner", private one to _actually_ only deliver the keys of child nodes, recursively.